compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/model/types/BXMLType.java

```diff
@@ -26,8 +27,26 @@
  */
 public class BXMLType extends BBuiltInRefType {
 
+    public BType constraint;
+    public boolean defaultConstraint;
```

Comment:
What's the purpose of the default constraint?

Reply:
The purpose of this to minimize the verbose xml error message for default xml type without any constraint. 
Please check toString() method of BXMLType.

Reply:
Can we remove `defaultConstraint` param from constructor (it's almost `false`) and check size == 4 in toString method.
If I'm not mistaken .toString will only be called when emitting diagnostics, slowness in .toString methods wouldn't affect much.

Reply:
+1 changed it to look for size. 
I am checking if `constraint instance of BUnionType` to avoid spotbugs warning eventhough we don't need it since we check typetags.